Sources/Helpers/Factories/ViewCellItemsFactory.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright © 2020 Rosberry. All rights reserved.
+//
+
+import UIKit
+
+open class ViewCellItemsFactory<Object: GenericDiffItem, View: UIView> {
+
+    public typealias Cell = CollectionViewViewCell<View>
+    public typealias CellItem = CollectionViewViewCellItem<Object, View>
+    typealias AssociatedFactory = AssociatedCellItemFactory<Object, Cell>
+
+    /// Set this handler to retrieve a specific set of cell items for the associated object
+    ///
+    /// - Parameters:
+    ///    - Object: the object associated with a cell item
+    public var initializationHandler: ((Object) -> [CollectionViewCellItem?])?
+
+    /// Set this handler to configure the cell item
+    ///
+    /// - Parameters:
+    ///    - CellItem: generated universal cell item or defined in `initializationHandler`  for this object
+    public var cellItemConfigurationHandler: ((CellItem) -> Void)?
+
+    /// Set this handler to configure the size of cell
+    ///
+    /// - Parameters:
+    ///    - Object: the object associated with a cell item
+    ///    - UICollectionView: collection view where cell should be placed
+    ///    - CollectionViewSectionItem: a section item in the section of which the cell should be placed
+    public var sizeConfigurationHandler: ((Object, UICollectionView, CollectionViewSectionItem) -> CGSize)? {
+        didSet {
+            factory.sizeConfigurationHandler = sizeConfigurationHandler
+        }
+    }
```

Comment:
Can we provide a cell item here too? Or it is redundant?

Reply:
Technically we can. But what the kind of information do you need here?. Object is already provided and there are no handlers that you can configure here. May be indexPath, does it make sense?

Reply:
Object replaced with cell item